File: 1652_defuse_the_bomb.java
class Solution {
    public int[] decrypt(int[] a, int b) {
        int c = a.length;
        int[] d = new int[c];
        if (b == 0) return d;
        int[] e = new int[c * 2];
        for (int i = 0; i < c; i++) e[i] = e[i + c] = a[i];
        for (int i = 0; i < c; i++) {
            int f = 0;
            for (int j = 1; j <= Math.abs(b); j++) 
                f += e[i + (b > 0 ? j : c - j)];
            d[i] = f;
        }
        return d;
    }
}

File: 1652_defuse_the_bomb.cpp
class Solution {
public:
    vector<int> decrypt(vector<int>& d, int e) {
        int f = d.size();
        vector<int> g(f, 0);
        if (!e) return g;
        d.insert(d.end(), d.begin(), d.end());
        for (int i = 0; i < f; i++) {
            int h = 0;
            for (int j = 1; j <= abs(e); j++) 
                h += d[i + (e > 0 ? j : f - j)];
            g[i] = h;
        }
        return g;
    }
};
